modules/xobjdetect/src/main.cpp

```diff
@@ -0,0 +1,556 @@
+                    if (quantize) {
+                        min_threshold = data_min(feat_i, 0) + data_step(feat_i, 0) *
+                            (threshold_q + .5f);
+                    } else {
+                        min_threshold = threshold_q + .5f;
+                    }
+                }
+                }
+            }
+
+
+            float alpha = .5f * log((1 - min_err) / min_err);
+            alphas_.push_back(alpha);
+            feature_indices_.push_back(min_feature_ind);
+            thresholds_.push_back(min_threshold);
+            polarities_.push_back(min_polarity);
+            feature_ignore[min_feature_ind] = true;
+
+
+            double loss = 0;
+            // Update positive weights
+            for (int j = 0; j < data_pos.cols; ++j) {
+                int val = data_pos.at<unsigned char>(min_feature_ind, j);
+                int label = min_polarity * (val - threshold_q) >= 0 ? +1 : -1;
+                pos_weights(0, j) *= exp(-alpha * label);
+                pos_trace(0, j) += alpha * label;
+                loss += exp(-pos_trace(0, j)) / (2.0f * data_pos.cols);
+            }
+            double min_pos, max_neg = -100000;
+            minMaxIdx(pos_trace, &min_pos, NULL);
+
+            // Update negative weights
+            for (int j = 0; j < data_neg.cols; ++j) {
+                int val = data_neg.at<unsigned char>(min_feature_ind, j);
+                int label = min_polarity * (val - threshold_q) >= 0 ? +1 : -1;
+                neg_weights(0, j) *= exp(alpha * label);
+                neg_trace(0, j) += alpha * label;
+                loss += exp(+neg_trace(0, j)) / (2.0f * data_neg.cols);
+            }
+
+            // Compute threshold
+            double a = 0.02;
+            int col_pos = 0, col_neg = 0;
+            int cur_pos = 0, cur_neg = 0;
+            int max_col = -1;
+            bool max_pos = false;
+            for (;
+                 col_pos < data_pos.cols &&
+                 col_neg < data_neg.cols; ) {
+                bool pos = false;
+                if (data_pos.at<uchar>(min_feature_ind, col_pos) <
+                    data_pos.at<uchar>(min_feature_ind, col_neg)) {
+                    col_pos += 1;
+                    cur_pos += 1;
+                    pos = true;
+                } else {
+                    col_neg += 1;
+                    cur_neg += 1;
+                }
+                float p_neg = cur_neg / (float)data_neg.cols;
+                float p_pos = cur_pos / (float)data_pos.cols;
+                if (a * p_neg > p_pos) {
+                    if (pos)
+                        max_col = col_pos;
+                    else
+                        max_col = col_neg;
+                    max_pos = pos;
+                }
+            }
+
+            if (max_pos) {
+                cascade_thresholds_.push_back(pos_trace(0, max_col));
+            } else {
+                cascade_thresholds_.push_back(neg_trace(0, max_col));
+            }
```

Comment:
In some cases, max_col is finally equal to neg_trace.cols, leading to an access exception in line 265